sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveDataFrameDropColumns.scala

```diff
@@ -36,7 +36,7 @@ class ResolveDataFrameDropColumns(val catalogManager: CatalogManager)
       //   df.drop(col("non-existing-column"))
       val dropped = d.dropList.map {
         case u: UnresolvedAttribute =>
-          resolveExpressionByPlanChildren(u, d.child)
+          resolveExpressionByPlanChildren(u, d)
```

Comment:
This is a good catch! When do we start to have this bug?

Reply:
Addressed rest comments. The bug was present when this rule was added.

Reply:
I have simply checked there is no more this buggy pattern for `resolveExpressionByPlanChildren` in code.